src/phoenix/server/api/types/Span.py

```diff
@@ -303,3 +343,21 @@ def _convert_metadata_to_string(metadata: Any) -> Optional[str]:
         return json.dumps(metadata)
     except Exception:
         return str(metadata)
+
+
+@dataclass
```

Comment:
```suggestion
@dataclass(frozen=True)
```

Suggestion: Immutable is a safer default.

Reply:
👍🏻

Reply:
Actually I can't do this as it doesn't comply with the HasSpanIO protocol